crates/red_knot_workspace/tests/check.rs

```diff
@@ -24,16 +24,26 @@ fn corpus_no_panic() -> anyhow::Result<()> {
     let corpus = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("resources/test/corpus");
     let system_corpus =
         SystemPathBuf::from_path_buf(corpus.clone()).expect("corpus path to be UTF8");
+    let stub_dir = tempfile::TempDir::new()?;
```

Comment:
I see the value in extending all corpus tests to `pyi` files and it kind of works but I could see this break in the future because the `stub_dir` isn't part of the workspace. This might cause problems in the future when we try to resolve settings because we won't have any settings available for files outside the workspace. 

I don't have a good suggestion on how to solve this differently other than copying a handful of tests and manually rename the files to `.pyi` files.

Reply:
I assume we will always need to support someone opening an out-of-workspace file? We just won't be able to resolve relative imports from such files. Which shouldn't be an issue for corpus tests. It might actually be useful to have a check that this doesn't break?

But if you'd rather, I can change the test so that it copies all the `.py` files to the tempdir also, and sets the tempdir as the workspace root?

Reply:
> I assume we will always need to support someone opening an out-of-workspace file? 

Yes, but it still needs to be clarified how we want to support this in the editor. One option is to create an ad-hoc workspace for all detached files. Another option is to list the detached files in the workspace explicitly. The issue with just using the workspace is that any detached file would automatically inherit the workspace setting. That might or might not be what we want. I don't see this use case in the CLI. 


> But if you'd rather, I can change the test so that it copies all the .py files to the tempdir also, and sets the tempdir as the workspace root?

That's a good idea. I would prefer this.